modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/object_entry/ScheduleField.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayCheckbox} from '@clayui/form';
+import {DatePicker} from '@liferay/object-js-components-web';
+import React, {useCallback, useEffect, useState} from 'react';
+
+interface ScheduleFieldProps {
+	checkboxLabel: string;
+	dateLabel: string;
+	error?: string;
+	id: string;
+	isChecked: boolean;
+	onCheckboxChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+	onDateChange: (value: string) => void;
+	portletNamespace: string;
+	value: string;
+}
+
+export default function ScheduleField({
+	checkboxLabel,
+	dateLabel,
+	id,
+	isChecked,
+	onCheckboxChange,
+	onDateChange,
+	portletNamespace,
+	value,
+}: ScheduleFieldProps) {
+	const [dateError, setDateError] = useState<string>('');
+
+	const [checked, setChecked] = useState<boolean>(isChecked);
+
+	const handleError = useCallback(
+		(value: string) => {
+			if (!value && !checked) {
+				setDateError(Liferay.Language.get('this-field-is-required'));
+			}
+			else {
+				setDateError('');
+			}
+		},
+		[checked]
+	);
+
+	useEffect(() => {
+		const saveButton = document.getElementById(
+			portletNamespace + 'saveObjectEntryButton'
+		);
+
+		if (!saveButton) {
+			return;
+		}
+
+		const handleClick = () => {
+			handleError(value);
+		};
+
+		saveButton.addEventListener('click', handleClick);
+
+		return () => {
+			saveButton.removeEventListener('click', handleClick);
+		};
+	}, [checked, handleError, portletNamespace, value]);
```

Comment:
I think you can take `checked` off the dependency array because `handleError` update will already take care of triggering the `useEffect`. Can you try this? I am afraid that having `checked` in both dependency arrays will trigger the `useEffect` twice when `checked` changes. Since this `useEffect` exists only for handling error, I guess having only `handleError` in the dependency array is enough, but we need to try it. Have you tried this? Thanks!

Reply:
Mabe React is smart enough to not trigger it twice, but still, having `checked` in both dependency arrays looked not-right to me since `checked` is not a direct dependency of the `useEffect`.

Reply:
you're absolutely right, thank you!! I just did that 😊